File: backend/app/schemas/overview.py
from pydantic import BaseModel


class SiteMonitorSummary(BaseModel):
    id: int
    name: str
    target: str
    is_up: bool
    response_ms: float | None = None


class SiteMonitorOverview(BaseModel):
    total: int
    up: int
    down: int
    items: list[SiteMonitorSummary]


class DomainCertSummary(BaseModel):
    id: int
    domain: str
    ssl_expired: bool
    days_remaining: int | None = None


class DomainCertOverview(BaseModel):
    total: int
    valid: int
    expiring: int
    expired: int
    items: list[DomainCertSummary]


class DomainWhoisSummary(BaseModel):
    id: int
    domain: str
    days_remaining: int | None = None


class DomainWhoisOverview(BaseModel):
    total: int
    valid: int
    expiring: int
    expired: int
    items: list[DomainWhoisSummary]


class MonitorSummaryResponse(BaseModel):
    site_monitors: SiteMonitorOverview
    domain_certs: DomainCertOverview
    domain_whois: DomainWhoisOverview

File: frontend/src/api/overview.ts
import { api } from './client'

export interface SiteMonitorSummary {
  id: number
  name: string
  target: string
  is_up: boolean
  response_ms: number | null
}

export interface SiteMonitorOverview {
  total: number
  up: number
  down: number
  items: SiteMonitorSummary[]
}

export interface DomainCertSummary {
  id: number
  domain: string
  ssl_expired: boolean
  days_remaining: number | null
}

export interface DomainCertOverview {
  total: number
  valid: number
  expiring: number
  expired: number
  items: DomainCertSummary[]
}

export interface DomainWhoisSummary {
  id: number
  domain: string
  days_remaining: number | null
}

export interface DomainWhoisOverview {
  total: number
  valid: number
  expiring: number
  expired: number
  items: DomainWhoisSummary[]
}

export interface MonitorSummary {
  site_monitors: SiteMonitorOverview
  domain_certs: DomainCertOverview
  domain_whois: DomainWhoisOverview
}

export function fetchMonitorSummary() {
  return api.get<MonitorSummary>('/overview/monitor-summary')
}
